src/EventStore.Common/Log/ThrottledLog.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Threading;
+using Serilog;
+using Serilog.Events;
+
+namespace EventStore.Common.Log; 
+
+public class ThrottledLog<T> {
+
+	private readonly ILogger _log;
+	private readonly long _duration;
+	private long _lastLogged;
+	public ThrottledLog(TimeSpan duration) {
+		_log = Serilog.Log.ForContext<T>();
+		_duration = duration.Ticks;
+		_lastLogged = DateTime.UnixEpoch.Ticks;
+	}
+
+	public bool Warning(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+	
+	public bool Fatal(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+	
+	public bool Information(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+	
+	public bool Error(string message) {
+		bool canLog = CanLog();
+		if (canLog) _log.Warning(message);
+		return canLog;
+	}
+
+	private bool CanLog() {
+		var currentTime = DateTime.Now.Ticks;
+		bool canLog = false;
+
+		// double-checked locking
+		if (currentTime - _lastLogged >= _duration) {
+			lock (_log) {
```

Comment:
Should the lock on `_log` be taken outside of the above `if` statement? `_lastLogged` is used in the condition but may change between checking it, taking the lock, and then updating it later

Reply:
Ah sorry, I missed that second check there, thanks!